include/mqt-core/dd/Operations.hpp

```diff
@@ -34,83 +34,99 @@
-                       const qc::Controls& controls, const qc::Qubit target,
-                       const bool inverse) {
-  GateMatrix gm;
 
-  const auto type = op->getType();
+inline std::pair<qc::OpType, std::vector<fp>>
+getOperationParameters(const qc::StandardOperation* op, qc::Qubit* target0,
+                       qc::Qubit* target1, bool inverse) {
+  auto type = op->getType();
   const auto& parameter = op->getParameter();
 
   switch (type) {
+  // self-inverse Operations
   case qc::I:
-    gm = I_MAT;
-    break;
   case qc::H:
-    gm = H_MAT;
-    break;
   case qc::X:
-    gm = X_MAT;
-    break;
   case qc::Y:
-    gm = Y_MAT;
-    break;
   case qc::Z:
-    gm = Z_MAT;
-    break;
+  case qc::SWAP:
+  case qc::ECR:
+    return {type, {}};
+  // inverse by using the inverse type
+  case qc::iSWAP:
+  case qc::iSWAPdg:
+  case qc::Peres:
+  case qc::Peresdg:
   case qc::S:
-    gm = inverse ? SDG_MAT : S_MAT;
-    break;
   case qc::Sdg:
-    gm = inverse ? S_MAT : SDG_MAT;
-    break;
   case qc::T:
-    gm = inverse ? TDG_MAT : T_MAT;
-    break;
   case qc::Tdg:
-    gm = inverse ? T_MAT : TDG_MAT;
-    break;
   case qc::V:
-    gm = inverse ? VDG_MAT : V_MAT;
-    break;
   case qc::Vdg:
-    gm = inverse ? V_MAT : VDG_MAT;
-    break;
-  case qc::U:
-    gm = inverse ? uMat(-parameter[1U], -parameter[2U], -parameter[0U])
-                 : uMat(parameter[2U], parameter[1U], parameter[0U]);
-    break;
-  case qc::U2:
-    gm = inverse ? u2Mat(-parameter[0U] + PI, -parameter[1U] - PI)
-                 : u2Mat(parameter[1U], parameter[0U]);
-    break;
-  case qc::P:
-    gm = inverse ? pMat(-parameter[0U]) : pMat(parameter[0U]);
-    break;
   case qc::SX:
-    gm = inverse ? SXDG_MAT : SX_MAT;
-    break;
   case qc::SXdg:
-    gm = inverse ? SX_MAT : SXDG_MAT;
-    break;
+    if (inverse) {
+      type = static_cast<qc::OpType>(+type ^ qc::OpTypeInv);
+    }
+    return {type, {}};
+  // inverse by negating the parameter
+  case qc::RXX:
+  case qc::RYY:
+  case qc::RZZ:
+  case qc::RZX:
   case qc::RX:
-    gm = inverse ? rxMat(-parameter[0U]) : rxMat(parameter[0U]);
-    break;
   case qc::RY:
-    gm = inverse ? ryMat(-parameter[0U]) : ryMat(parameter[0U]);
-    break;
   case qc::RZ:
-    gm = inverse ? rzMat(-parameter[0U]) : rzMat(parameter[0U]);
-    break;
+  case qc::P:
+    if (inverse) {
+      return {type, {-parameter[0U]}};
+    }
+    return {type, {parameter[0U]}};
+  // inverse by negating the first of two parameter
+  case qc::XXminusYY:
+  case qc::XXplusYY:
+    if (inverse) {
+      return {type, {-parameter[0U], parameter[1U]}};
+    }
+    return {type, {parameter[0U], parameter[1U]}};
+
+  case qc::DCX:
+    if (inverse) {
+      if (target0 == nullptr || target1 == nullptr) {
+        throw qc::QFRException("Invalid target qubits for DCX");
+      }
+
+      // DCX is not self-inverse, but the inverse is just swapping the targets
+      std::swap(*target0, *target1);
+    }
```

Comment:
Feels a bit strange that this is the only case in all of this function that actually needs the `target0` and `target1` parameters.
That triggered a bit of a thought process.

Maybe a slight optimization for this overall method would be to pass the targets as a `std::vector<Qubit>&` as well. That way, 
- it can contain various entries which makes it more future proof
- only requires 8 Byte for the parameter (instead of 16 Byte for the two pointers)
- directly allows to pipe-through the return value of the `getTargets()` method in the `Operation` class.
- should simplify some of the calls across the library slightly.

This would also allow one to not distinguish between the single-qubit and the two-qubit case all that much and have a single function that takes a vector of targets and handles the distinction internally.
But maybe this is also a stupid idea. It be nice if you could play around a little with this.